lib/blocktree/blocktree.go

```diff
@@ -149,8 +153,24 @@ func (bt *BlockTree) GetAllBlocksAtDepth(hash common.Hash) []common.Hash {
 	return bt.head.getNodesWithDepth(depth, hashes)
 }
 
+func (bt *BlockTree) setInCache(b *node) {
+	if b == nil {
+		return
+	}
+
+	if _, has := bt.nodeCache[b.hash]; !has {
+		bt.nodeCache[b.hash] = b
+	}
+}
+
 // getNode finds and returns a node based on its Hash. Returns nil if not found.
-func (bt *BlockTree) getNode(h Hash) *node {
+func (bt *BlockTree) getNode(h Hash) (ret *node) {
+	defer func() { bt.setInCache(ret) }()
```

Comment:
```suggestion
	defer bt.setInCache(ret)
```

Reply:
Correct me if i'm wrong but wouldn't that cause the `ret` param to be evaluated as `nil`, because it is immediatly evaluated in the `defer` call and not when the function returns??

Reply:
interesting, yeah looks like that is the case. this is good then!